photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/CoordinateDescent.scala

```diff
@@ -178,35 +189,47 @@ class CoordinateDescent(
             s"iteration $iteration is:\n$objectiveFunctionValue")
 
         // Update the validating score and evaluate the updated model on the validating data
-        validatingScoresContainerOption = validatingDataAndEvaluatorsOption.map { case (validatingData, evaluators) =>
-          var validatingScoresContainer = validatingScoresContainerOption.get
-          val validatingScores = updatedModel
-            .score(validatingData)
-            .setName(s"Updated validating scores with coordinateId $coordinateId")
-            .persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
-            .materialize()
-          validatingScoresContainer(coordinateId).unpersistRDD()
-          validatingScoresContainer = validatingScoresContainer.updated(coordinateId, validatingScores)
-          val fullScore = validatingScoresContainer.values.reduce(_ + _)
-          evaluators.foreach { evaluator =>
-            val validationTimer = Timer.start()
-            val evaluationMetric = evaluator.evaluate(fullScore.scores)
-            logger.debug(s"Finished validating the model, time elapsed: ${validationTimer.stop().durationSeconds} (s).")
-            logger.info(s"Evaluation metric computed with ${evaluator.getEvaluatorName} after updating " +
-                s"coordinateId $coordinateId at iteration $iteration is $evaluationMetric")
-          }
-          validatingScoresContainer
+        // if there are evaluators defined
+        val currentEval = validatingDataAndEvaluatorsOption.map {
+          case (validatingData, evaluators) =>
+            validatingScoresContainerOption.get.updated(coordinateId,
+              updatedModel
+                .score(validatingData)
+                .setName(s"Updated validating scores for coordinateId $coordinateId")
+                .persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
+                .materialize()
+            )
```

Comment:
This looks like we're not really updating `validatingScoresContainerOption` here. We call `updated`, but then drop the updated value completely. This will cause subsequent coordinates to compute an incorrect ```fullScore```.